app/views.py

```diff
@@ -1,23 +1,94 @@
 from flask import Flask, request, jsonify
 from flask_restful import Api, Resource
 import json
+import random
+
+from .models import User, Requests, Requests_Schema
 
 app = Flask(__name__)
 api = Api(app)
 
-class UserAPI(Resource):
+users = []
+requests = []
+
+class UserSignupAPI(Resource):
+    def post(self):
+        user = request.get_json()
+        username = user['username']
+        department = user['department']
+        email = user['email']
+        password = user['password']
+
+        u = User(
+            username=username,
+            department=department,
+            email=email,
+            password=password
+            )
+    
+        if not user['username'] or not user['department'] or not user['email'] \
+                or not user['password']:
+            return {'message': 'All fields required'}, 400
+            
+        elif u.email in [i.email for i in users]:
+            return {"message": "User already exists"}, 203
+    
+        elif u.username in [i.username for i in users]:
+            return {"message": "User already exists"}, 203
+
+        users.append(u)
+
+        return {"message":"Successfully registered"},201 
+
+class UserLoginAPI(Resource):
+
     def post(self):
-        pass
+        access = request.get_json()
+        username = access['username']
```

Comment:
When accessing indices of a dictionary that may be missing use the `get` method e.g access.get('username). This will not throw an exception and will return `None` if the key is missing. You can then check `if not username` which will behave the same way for an empty string and a NoneType object.